pdb-slave/MARCH4-PDB/lib/StateMachine/StateMachine.cpp

```diff
@@ -0,0 +1,184 @@
+#include "StateMachine.h"
+
+// Constructor
+StateMachine::StateMachine()
+{
+   this->currentState = Init_s;
+   this->onOffButtonTimer.reset();
+   this->onOffButtonTimer.start();
+   this->ledTimer.reset();
+   this->onOffButtonLedState = false;
+   this->keepPDBOn = false;
+   this->LVon = false;
+   this->masterShutdown = false;
+}
+
+// Updates the current state based on the button inputs
+void StateMachine::updateState(bool buttonState, bool masterOkState, bool shutdownAllowedState){
+    int buttonTimeMs = this->onOffButtonTimer.read_ms();
+    int ledTimeMs = this->ledTimer.read_ms();
+    switch(this->currentState){
+        case Init_s:
+            this->masterShutdown = false;
+            if(buttonTimeMs > this->onOffButtonTimeShort){
+                this->onOffButtonLedState = true;
+                this->keepPDBOn = true;
+            }
+            if(!buttonState){
+                if(buttonTimeMs <= this->onOffButtonTimeShort){
```

Comment:
For readability you could put this condition in a method. You are using it at least 3 times